### PR TITLE
Bump the version of Go

### PR DESCRIPTION
**Pull request by mattmoor**
ref: https://github.com/sigstore/fulcio/pull/1703#discussion_r1649450336

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,6 +1,6 @@
 module github.com/sigstore/fulcio
 
-go 1.21.9
+go 1.22.2
```
Comment (haydentherapper) [inline go.mod:3]: Bump to the latest, 1.22.4?

**Comment by haydentherapper** [comment]
@cpanato can you take a look at if any of the GHAs need to be updated as well?

File: go.mod (hunk 1 above)
Comment (mattmoor) [reply go.mod:3]:
```suggestion
go 1.22.4
```

**Comment by cpanato** [comment]
update this pr here: https://github.com/sigstore/fulcio/pull/1707

**Comment by cpanato** [comment]
closing in favor of https://github.com/sigstore/fulcio/pull/1707

**Pull request closed without merging**